### PR TITLE
Cleanup some System.Reflection.Emit.ILGeneration tests

### PR DESCRIPTION
**Pull request by hughbe**
- Take advantage of xunit
- Consolidate some really short tests into their own file
- Deobfuscate these tests and rename them

/cc @AtsushiKan

### CODE REVIEW

File: src/System.Reflection.Emit.ILGeneration/tests/CustomAttributeBuilder/CustomAttributeBuilderCtor2.cs [hunk 1]
```diff
@@ -2,984 +2,194 @@
...
-
-        public CustomAttributeBuilderTest(string testString, int testInt32, string getOnlyString, int getOnlyInt32)
-        {
-            this.TestStringField = testString;
-            this.TestInt = testInt32;
-            this.GetString = getOnlyString;
-            this.GetInt = getOnlyInt32;
-        }
-    }
+    using static Helpers;
```
Comment (stephentoub) [inline src/System.Reflection.Emit.ILGeneration/tests/CustomAttributeBuilder/CustomAttributeBuilderCtor2.cs]: I'd prefer we not use this.  These tests are sprawling, with lots of test class and lots of files, and seeing an unadorned method call suggests that the method is in that test class or a base or something.  It doesn't save much typing, and without it it's very clear where those helpers are coming from.

Comment (hughbe) [inline src/System.Reflection.Emit.ILGeneration/tests/CustomAttributeBuilder/CustomAttributeBuilderCtor2.cs]: You're right. I've fixed this

**Comment by AtsushiKan** [comment]
LGTM.

**Comment by stephentoub** [comment]
Thanks, @hughbe.

**Pull request merged**